src/components/panel/panel.js

```diff
@@ -828,11 +825,15 @@ MdPanelRef.prototype.open = function() {
   // TODO(ErinCoughlan) - Cancel any in-progress actions.
 
   var self = this;
+
+  var onOpenComplete = self._config['onOpenComplete'];
   this._openPromise = this._$q(function(resolve, reject) {
     self.attachOnly().then(function() {
-      self.show().then(function() {
-        resolve(self);
-      }, reject);
+      self.show()
+          .then(self._$q.when(self._config['scope'].$eval(onOpenComplete)), reject)
```

Comment:
Please move this to the end of the show() section. If the user calls attach and show on their own we still want this callback to trigger.